modules/data_release/ajax/FileUpload.php

```diff
@@ -21,47 +21,55 @@
     $upload_date = date('Y-m-d');
     $base_path   = __DIR__ . "/../user_uploads/";
 
-    $target_path = $base_path . $fileName;
-    if (move_uploaded_file($_FILES["file"]["tmp_name"], $target_path)) {
-        $success = $DB->insert(
-            'data_release',
-            array(
-             'file_name'   => $fileName,
-             'version'     => $version,
-             'upload_date' => $upload_date,
-            )
-        );
-
-        $user_ID = $DB->pselectOne(
-            "SELECT ID FROM users WHERE userid=:UserID",
-            array('UserID' => $user->getUsername())
-        );
-        $ID      = $DB->pselectOne(
-            "SELECT id FROM data_release WHERE "
-            . "file_name=:file_name AND "
-            . "version=:version AND "
-            . "upload_date=:upload_date",
-            array(
-             'file_name'   => $fileName,
-             'version'     => $version,
-             'upload_date' => $upload_date,
-            )
-        );
-        $success = $DB->insert(
-            'data_release_permissions',
-            array(
-             'userid'          => $user_ID,
-             'data_release_id' => $ID,
-            )
-        );
-    }
-
     $factory  = NDB_Factory::singleton();
     $settings = $factory->settings();
 
     $baseURL = $settings->getBaseURL();
 
-    header("Location: {$baseURL}/data_release/?uploadSuccess=true");
+    if (!file_exists(__DIR__ . "/../user_uploads/")) {
+        header("HTTP/1.1 400 Bad Request");
+        echo "File upload failed. Default user_uploads directory not found.";
+    } elseif (!is_writable(__DIR__ . "/../user_uploads/")) {
+        header("HTTP/1.1 400 Bad Request");
```

Comment:
These seem to be server configuration errors (500), not client errors (400).

They also seem like they're exceptional circumstances since they only happen on misconfigured servers. Is there any reason to not just throw an exception?